### PR TITLE
Improve caret version constraint documentation

### PR DESCRIPTION
**Pull request by cereblanco**
# Pull Request Check List

Add more context to the sentence `In this case, if we ran poetry update requests...`

Resolves: https://github.com/python-poetry/poetry/issues/3242

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by salty-horse** [comment]
The library name in the example was changed from `requests` to `request`. Is this on purpose?

File: docs/docs/dependency-specification.md [hunk 1]
```diff
@@ -7,12 +7,7 @@ of the dependency and on the optional constraints that might be needed for it to
 
 ### Caret requirements
 
-**Caret requirements** allow SemVer compatible updates to a specified version.
-An update is allowed if the new version number does not modify the left-most non-zero digit in the major, minor, patch grouping.
-In this case, if we ran `poetry update requests`, poetry would update us to version `2.14.0` if it was available,
-but would not update us to `3.0.0`.
-If instead we had specified the version string as `^0.1.13`, poetry would update to `0.1.14` but not `0.2.0`.
-`0.0.x` is not considered compatible with any other version.
+**Caret requirements** allow [SemVer](https://semver.org/) compatible updates to a specified version. An update is allowed if the new version number does not modify the left-most non-zero digit in the major, minor, patch grouping. For instance, if we previously ran `poetry add request@^2.13.0`, and wanted to update the library and ran `poetry update request`, poetry would update us to a latest version say `2.14.0`, but would not update us to `3.0.0`. If instead we had specified the version string as `^0.1.13`, poetry would update to `0.1.14` but not `0.2.0`. `0.0.x` is not considered compatible with any other version.
```
Comment (salty-horse) [inline docs/docs/dependency-specification.md:10]: > poetry would update us to a latest version say `2.14.0`

This was fine before, and shouldn't change:
>poetry would update us to version `2.14.0` if it was available

**Comment by cereblanco** [comment]
> The library name in the example was changed from `requests` to `request`. Is this on purpose?

@salty-horse Ah! It was not on purpose. I reverted it. 
Also, I reverted the old phrase `poetry would update us to version 2.14.0 if it was available`.

Thanks! 😺

**Review by finswimmer — APPROVED** [review]
Thanks a lot :+1:

**Pull request merged**